include/jemalloc/internal/prof_structs.h

```diff
@@ -201,4 +208,15 @@ struct prof_tdata_s {
 };
 typedef rb_tree(prof_tdata_t) prof_tdata_tree_t;
 
+struct prof_recent_s {
+	nstime_t alloc_time;
+	nstime_t dalloc_time;
+
+	prof_recent_t *next;
+	size_t usize;
+	prof_tctx_t *alloc_tctx;
+	extent_t *alloc_extent; /* NULL means allocation has been freed. */
```

Comment:
Are the prof_recent <=> extent pointing to each other used only for tracking freed_or_not?

Reply:
It's more than a boolean flag -

First, the allocation-to-record pointer is the only way for the allocation to find the record when the allocation is being freed, so that information like deallocation time / thread / back trace can be recorded there, in addition to the deallocation event itself.

Then, there comes the need for the record-to-allocation pointer, which serves to reset the allocation-to-record pointer on the allocation whenever the record dies.